frontend/src/tests/utils.js

```diff
@@ -2,10 +2,41 @@ import axios from "axios";
 import PropTypes from "prop-types";
 import { apiPropTypes } from "../api/defs/prop-types";
 // eslint-disable-next-line
-const { expect, test } = global;
+const { expect, test, it, describe, beforeAll, afterAll } = global;
 
-// Jest demands there be a test in every file, so we put a no-op test here.
-test("no-op", () => {});
+// add a custom `.toContainObject` method to `expect()` to see if an array contains
+// an object with matching props. Taken from
+// https://medium.com/@andrei.pfeiffer/jest-matching-objects-in-array-50fe2f4d6b98
+expect.extend({
+    toContainObject(received, argument) {
```

Comment:
Good point but this requires a package called `jest-extended` which I am not sure if we have it or not.

Reply:
No I don't think you do. I've been playing around with it and I was able to use it without that package

Reply:
Then, in this case, we have the package somewhere (probably with the latest create react app or jest). But according to the link you attached and the GitHub page it leads to, it's a separate package of jest

Reply:
I wasn't using it in our app, just doing some simple jest testing separately. Are you talking about where it says _"For additional Jest matchers maintained by the Jest Community check out jest-extended"_ ? If you follow that link it looks like there are more matchers in addition to the ones included in the link I provided that aren't maintained by jest by by the jest community. The link I provided is the official jest documentation so I believe `toContainEqual()` and all the other ones there are provided without any other packages

Reply:
[https://jestjs.io/docs/en/expect#tocontainequalitem](url) Try copying and pasting this link into your browser. When I click it directly it leads to a 404 page not found on Github. But if you copy and paste it leads you to the page I was talking about (official jest docs)

Reply:
Yeah, I opened the correct link. It doesn't matter since it works. (i was talking about the README on this page [https://github.com/jest-community/jest-extended])

Reply:
since it works, the next question would be: are they actually do the same thing.(I have read it carefully, will do that later)

Reply:
they are slightly different tho:
```javascript
const myBeverage = { delicious: true };
const l = [{ delicious: true, sour: false }];

expect(l).toContainEqual(myBeverage); // fail
expect(l).toContainObject(myBeverage); // pass
```